lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -27,9 +27,21 @@ class AwsCompileCloudFrontEvents {
       'before:remove:remove': this.logRemoveReminder.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8025
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'cloudFront', {
       type: 'object',
+      properties: {
+        eventType: {
+          type: 'string',
+          enum: ['viewer-request', 'origin-request', 'origin-response', 'viewer-response'],
+        },
+        pathPattern: { type: 'string', pattern: '^([A-Za-z0-9-_.*$/~"\'@:+]|&amp;)+$' },
```

Comment:
`-` char needs to be escaped (otherwise in `[]` it implies a range between characters.

Reply:
moved it to the end of the array like we do in other patterns